### PR TITLE
WIP: Add a way to archive logs after each phase

### PR DESCRIPTION
**Pull request by drehak**

### CODE REVIEW

**Comment by bocekm** [comment]
What Jira task is this change related to, or, why is this needed?

**Comment by drehak** [comment]
> What Jira task is this change related to, or, why is this needed?

@bocekm It's related to saving logs from containers, after each phase, as Vinny suggested. (We still have to get the files out, this is just the framework part.)

File: leapp/utils/phaselogs.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import errno
+import os
+import tarfile
+
+from leapp.exceptions import CommandError
+
+
+def init_phase_logs(phase_logs_archive, logger=None):
```
Comment (ZhukovGreen) [inline leapp/utils/phaselogs.py:8]: What is the purpose of having the logger as an argument?

**Review by ZhukovGreen — COMMENTED** [review]
Hi @drehak, just a little comment here. Please provide some description of the merge request (even thought it is in WIP), so it will be easier to review then

**Comment by drehak** [comment]
> Hi @drehak, just a little comment here. Please provide some description of the merge request (even thought it is in WIP), so it will be easier to review then

@ZhukovGreen Hi, this is a very rough prototype mostly for getting builds I can use quickly. It's not meant for reviews yet.

**Comment by ZhukovGreen** [comment]
@drehak this is a qote from OAMG-3808:
```
1. Each feature might have more than one MR. These MRs are very small and they are referenced to the relevant feature ticket.  As a result we're forming a kind of a story from MRs, which describes how the developer was progressing on the feature (*story-telling by merge requests*).
 2. OAMG developers reviewing these small MRs on a daily basis. This allows developers to understand the context behind the particular MR, the author's rationale and expected target. As a result, this gives a different, a way better quality, code review

The closest example of this approach might be [this github ticket |https://github.com/oamg/leapp-repository/issues/561] . In our usual way, this feature (basic pre-commit setup) would be introduced by a one big MR, but in this example, there will be around 5-7 MRs. Each MR is very small, and only when all these MRs will be merged the ticket will be closed.
```

**Comment by ZhukovGreen** [comment]
@drehak so I mean, this is much better to show what you're trying to achieve and then start doing it in a very small steps

**Comment by drehak** [comment]
@ZhukovGreen Well, since this MR would already constitute one half of the final solution, I guess I could do it that way and link it to a new issue :) I just don't see the point in making it any more granular, since all the code in this MR ties into itself and with any parts out, the code would either be dead code, or broken.
Regardless, I'll change the commit message and description so it's mergeable. (It still needs unit tests though.)

**Comment by vojtechsokol** [comment]
failed in e2e tests:
```
2020-09-03 15:39:16.266 INFO     PID: 1947 leapp: Logging has been initialized
2020-09-03 15:39:16.268 INFO     PID: 1947 leapp: Initializing archive at /var/log/leapp/phase-logs.tar.gz
Traceback (most recent call last):
  File "/usr/bin/leapp", line 9, in <module>
    load_entry_point('leapp==0.11.0', 'console_scripts', 'leapp')()
  File "/usr/lib/python2.7/site-packages/leapp/cli/__init__.py", line 18, in main
    cli.command.execute('leapp version {}'.format(VERSION))
  File "/usr/lib/python2.7/site-packages/leapp/utils/clicmd.py", line 103, in execute
    args.func(args)
  File "/usr/lib/python2.7/site-packages/leapp/utils/clicmd.py", line 125, in called
    self.target(args)
  File "/usr/lib/python2.7/site-packages/leapp/cli/upgrade/__init__.py", line 230, in upgrade
    init_phase_logs(phase_logs_archive, logger=logger)
  File "/usr/lib/python2.7/site-packages/leapp/utils/phaselogs.py", line 33, in init_phase_logs
    with tarfile.open(phase_logs_archive, mode='x:gz') as tarball:  # noqa: F841
  File "/usr/lib64/python2.7/tarfile.py", line 1678, in open
    return func(name, filemode, fileobj, **kwargs)
  File "/usr/lib64/python2.7/tarfile.py", line 1713, in gzopen
    raise ValueError("mode must be 'r' or 'w'")
ValueError: mode must be 'r' or 'w'
```

**Comment by ZhukovGreen** [comment]
> @ZhukovGreen Still thinking about this:
> 
> > Each feature might have more than one MR. These MRs are very small and they are referenced to the relevant feature ticket. As a result we're forming a kind of a story from MRs, which describes how the developer was progressing on the feature (story-telling by merge requests).
> 
> 
> 
> Aren't individual commits already telling the story quite clearly? As far as I know, we have no "1 commit per 1 MR" rule and there can be multiple commits in a PR, merged to the master branch without squashing.

@drehak individual commits are good, but the code review is not assuming that people usually really pass through commits one by one. You just watch the diff with the master and this is it. Gitlab, actually, has this feature. But we are not on gitlab

**Comment by drehak** [comment]
Closing this in lieu of another prototype to start clean. (PR soon.)

**Pull request closed without merging**